src/ObjCRuntime/Protocol.cs

```diff
@@ -24,6 +24,18 @@ public Protocol (string name)
 				throw new ArgumentException (String.Format ("'{0}' is an unknown protocol", name));
 		}
 
+		public Protocol (Type type)
+		{
+			this.handle = Class.GetHandle (type);
```

Comment:
Mmh didn't realized I could dive into the managed type like that! Much better thanks.